tests/test_cache.py

```diff
@@ -0,0 +1,13 @@
+"""Tests of GDAL VSI cache invalidation."""
+
+from rasterio import cache
+
+
+def test_invalidate_all():
+    """Cache is entirely invalidated."""
+    cache.invalidate_all()
```

Comment:
It's a smoke test. I'm inclined to rely on GDAL's testing. `invalidate_all()` simply calls a GDAL method with no arguments. There's no way to use it wrongly. `invalidate()` uses the same path parsing pattern used elsewhere in rasterio. It's reliable.